### ISSUE TITLE
Problems parsing PROGRAM with TCP

### ISSUE THREAD

**Issue by jguillouf** [balabit/syslog-ng#1056]
Hi,

I have my syslog-ng server listen on UDP port but we lost many lines each days.

So I decided to send logs in TCP.

It works well but I have a strange behavior with one rule :
```
source s_all {
        udp(ip(10.0.0.14));
        tcp(ip(10.0.0.14) max_connections(50));
};

destination d_service {
        file("/var/log/test/$HOST/$YEAR/$MONTH/$DAY/$PROGRAM"
                create_dirs(yes)
        );
};

log { source(s_all); destination(d_service);  };
```

The rule works fine with UDP, it creates the expected files:

> - kernel
> - sshd
> - CRON
> - rsyslogd
> etc.

But with TCP it writes the same files but also some files like : 

> - 1_choice_0\"                                                                 
> - \">\n
> - 1e7172676h38195a_17_1"                                                          
> - 459edzqd7pl6a777_8_1_choice_2\"                                          
> - class=\"status                                                                
> - pqzdkqodzq-99001bloirc-problem-36d458sf78788d8d6sfsefee144afc4_4_1

I looks like HTML code but I don't understand why it creates these files with TCP.

The client configuration is the same... I just add a 2nd "@" to tell rsyslog to send TCP packets.

Version:
syslog-ng server : 3.3.4.dfsg-2ubuntu1
rsyslod clients : 5.8.6-1ubuntu8.7

Please tell me how to solve my problem.

Best regards,

**Comment by bazsi** [comment]
You should dump and inspect the traffic between rsyslog and syslog-ng. For
example wireshark can do that. Or record the traffic with tcpdump and
review it using wireshark on another computer.

This should work

**Comment by ihrwein** [comment]
What if a UDP datagram contains only one message while the messages are separated by newlines in case of TCP?

**Comment by bazsi** [comment]
Yeah, that's the expectation. When using tcp, newlines provide the framing
to identify individual elements. You can't rely on datagram boundaries in
the case of tcp.


-- 
Bazsi

**Comment by jguillouf** [comment]
Hi,

So here what is do : 
- I see that a file name **0e4aca830303a65c24682b_4_1_status\"** was created today at 11h39.
- I know that it's my "test_tracking" logs which are the problem
- I checked with tcpdump/wireshark , but in my case there is no newlines before  **0e4aca830303a65c24682b_4_1_status\"** .
- In my syslog server, I looked at the test_tracking.log file : 
-> there is an entry at 11:39 finished that way : 
> issue-679697j78979**0e4aca830303a65c24682b_4_1_status\"** size=\"20\"/>\n\n      <span class=\"status\" id=\"i4x-OrangeTelecom-69696969-
-> I discovered that the lines which contains morz than 8197 characters are truncated.

But I don't understand why it's truncated after 8197 chars.
